core/collector.class.inc.php

```diff
@@ -576,18 +577,33 @@ protected function AddHeader($aHeaders)
 	{
 		$this->aCSVHeaders = array();
 		foreach ($aHeaders as $sHeader) {
-			if (($sHeader != 'primary_key') && !array_key_exists($sHeader, $this->aFields)) {
+			if (($sHeader != 'primary_key') && !$this->HeaderIsAllowed($sHeader)) {
 				if (!$this->AttributeIsOptional($sHeader)) {
```

Comment:
Especially this part is key to allow for more customisation of the data collectors..